polkadot/xcm/xcm-executor/src/lib.rs

```diff
@@ -1282,4 +1273,37 @@ impl<Config: config::Config> XcmExecutor<Config> {
 				}),
 		}
 	}
+
+
+	/// Deposit `to_deposit` assets to `beneficiary`, without giving up on the first (transient) error, and
+	/// retrying once just in case one of the subsequently deposited assets satisfy some requirement.
+	///
+	/// Most common transient error is: `beneficiary` account does not yet exist and the first asset(s) in
+	/// the (sorted) list does not satisfy ED, but a subsequent one in the list does.
+	fn deposit_assets_with_retry(
+		&mut self,
+		to_deposited: &AssetsInHolding,
+		beneficiary: &Location,
+	) -> Result<(), XcmError> {
+		let mut failed_deposits = Vec::with_capacity(to_deposited.len());
+
+		for asset in to_deposited.assets_iter() {
+			let asset_result =
+				Config::AssetTransactor::deposit_asset(&asset, &beneficiary, Some(&self.context));
+			// if deposit failed for asset, mark it for retry after depositing the others
+			if asset_result.is_err() {
+				failed_deposits.push(asset);
+			}
+		}
+
```

Comment:
Maybe we could early return if `failed_deposits` length is equal to `to_deposit` length.
That would save one attempt.

Reply:
Changed! 
For this case, I'm returning the first error we received from the deposits.